service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapper.java

```diff
@@ -177,12 +177,12 @@ private String buildExpirationDatePertinentInformation(Immunization immunization
     }
 
     private String buildSitePertinentInformation(Immunization immunization) {
-        Optional<String> site = Optional.empty();
-        if (immunization.hasSite()) {
-            CodeableConcept siteObject = immunization.getSite();
-            site = CodeableConceptMappingUtils.extractTextOrCoding(siteObject);
-        }
-        return site.map(value -> SITE + value).orElse(StringUtils.EMPTY);
+        StringBuilder siteInformation = new StringBuilder(StringUtils.EMPTY);
+        CodeableConceptMappingUtils.extractUserSelectedTextOrCoding(immunization.getSite()).ifPresent(siteValue -> {
```

Comment:
Don't you need to check for `hasSite()` ?

Reply:
It's more readable when `ifPresent` is on new line

Reply:
Anyway why not do this:
```
return Optional.of(immunization)
    .filter(Immunization::hasSite)
    .map(Immunization::getSite)
    .map(siteValue -> SITE + siteValue)
    .orElse(StringUtils.EMPTY);
```

Reply:
Addressed